### ISSUE TITLE
Using set after loadHelper in beforeRender

### ISSUE THREAD

**Issue by leighmackay** [cakephp/cakephp#6993]
Hello,

In CakePHP v3.0.8 "set" only works when it's above "loadHelper"  in "beforeRender".

Does not work: set() is below loadHelper() 
```
public function beforeRender(Event $event) {
  $this->getView()->loadHelper('OrangeFingers');
  $cheezels = "numnumnum";
  $this->set(compact('cheezels'));
}
```

Works: set() is above loadHelper() 
```
public function beforeRender(Event $event) {
  $cheezels = "numnumnum";
  $this->set(compact('cheezels'));
  $this->getView()->loadHelper('OrangeFingers');
}
```

**Comment by leighmackay** [comment]
Just had another look at this, it's $this->getView() that is making this occur.

**Comment by ADmad** [comment]
This occurs because the view vars are passed from controller to View class only when view instance is created. The issue has been fixed in 3.1. Would you like to make a PR back porting the fix? It basically involves adding these 2 lines [61](https://github.com/cakephp/cakephp/blob/3.1/src/View/ViewVarsTrait.php#L61) and [85](https://github.com/cakephp/cakephp/blob/3.1/src/View/ViewVarsTrait.php#L85).

**Comment by dereuromark** [comment]
Closing as PR is now open.
